hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -464,4 +487,138 @@ private static Account getEffectiveOwnerAccount(
             return ownerAccount;
         }
     }
+
+    @NonNull
+    @Override
+    public Fees calculateFees(@NonNull final FeeContext feeContext) {
+        final var body = feeContext.body();
+        final var op = body.cryptoApproveAllowanceOrThrow();
+        final var accountStore = feeContext.readableStore(ReadableAccountStore.class);
+
+        final var txnValidStart =
+                body.transactionIDOrThrow().transactionValidStartOrThrow().seconds();
+        final var payerId = body.transactionIDOrThrow().accountIDOrThrow();
+        final var account = accountStore.getAccountById(payerId);
```

Comment:
Again, `getIfUsable(...)` needed here?

Reply:
yeah, but no access to ExpiryValidator here. Will add these checks in all `calculateFees` methods in my next PR